### PR TITLE
[Dev-1359] nginx header fix

### PR DESCRIPTION
**Pull request by jogendrajangid-tryjeeves**
Added nginx.conf using configmap

### CODE REVIEW

File: deploy/helm/templates/statefulset.yaml [hunk 1]
```diff
@@ -100,6 +102,10 @@ spec:
       - name: data
         emptyDir: {}
   {{- else }}
+      - name: nginx-config
+        configMap:
+          name: nginxconfig
+
```
Comment (marcocaberletti) [inline deploy/helm/templates/statefulset.yaml:108]: to mount a single file, without "cover" all the directory, I guess we should do something like:
```
- name: nginx-config
   configMap:
      name: nginxconfig
      items:
      - key: nginx.conf
         path: nginx.conf
```

File: deploy/helm/templates/statefulset.yaml [hunk 2]
```diff
@@ -88,6 +88,8 @@ spec:
           volumeMounts:
             - name: data
               mountPath: /appsmith-stacks
+            - name: nginx-config
+              mountPath: /etc/nginx/nginx.conf
```
Comment (marcocaberletti) [inline deploy/helm/templates/statefulset.yaml:92]: Add:
```
subPath: nginx.conf
readOnly: true
```
(the reason in the comment below)

**Review by marcocaberletti — CHANGES_REQUESTED** [review]

File: deploy/helm/templates/statefulset.yaml (hunk 1 above)
Comment (jogendrajangid-tryjeeves) [reply deploy/helm/templates/statefulset.yaml]: Done

File: deploy/helm/templates/nginx-configMap.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: nginxconfig
+  namespace: {{ include "appsmith.namespace" . }}
+  labels:
+    {{- include "appsmith.labels" . | nindent 4 }}
+data:
+  nginx.conf: |
+  user www-data;
```
Comment (marcocaberletti) [inline deploy/helm/templates/nginx-configMap.yaml:11]: fix indentation

**Review by marcocaberletti — CHANGES_REQUESTED** [review]

File: deploy/helm/templates/nginx-configMap.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  labels:
+    {{- include "appsmith.labels" . | nindent 4 }}
+data:
+  nginx.conf: |
+    user www-data;
+    worker_processes auto;
+    pid /run/nginx.pid;
+    include /etc/nginx/modules-enabled/*.conf;
+    events {
+    worker_connections 768;
```
Comment (marcocaberletti) [inline deploy/helm/templates/nginx-configMap.yaml:16]: fix format

**Review by marcocaberletti — CHANGES_REQUESTED** [review]
fix format

**Review by marcocaberletti — APPROVED** [review]

**Pull request merged**